### ISSUE TITLE
"IndexCompilationException: Cannot extract field name from: 00" when you use "default" for DateTime

### ISSUE THREAD

**Issue by oiBio** [ravendb/ravendb#15979]
Hello,

just stombeld over this error.
I used " = default" for a DateTime Property and get this exception on building the index.
Its an easy fix to use `DateTime.MinValue` instead, but i think that `default `should work also.

```
IndexCompilationException: IndexCompilationException: Cannot extract field name from: 00, IndexDefinitionProperty='Maps', ProblematicText='docs.Bewerber.Select(c => new {
    Working = ((string) null),
    WorkingInt = 0,
    NotWorking = 01 / 01 / 0001 00: 00: 00
})' ---> System.NotSupportedException: Cannot extract field name from: 00
   at Raven.Server.Documents.Indexes.Static.Roslyn.Rewriters.RewritersHelper.ExtractFields(AnonymousObjectCreationExpressionSyntax 
```

Here is the Index definition

```
public class TestIndex : AbstractIndexCreationTask<Entity, TestIndex.Result>
{
    public class Result
    {
        public string Working { get; set; }
        public int WorkingInt { get; set; }
        public DateTime NotWorking { get; set; }
    }
    public TestIndex()
    {
        Map = collection => from c in collection
                            select new Result
                            {
                                Working = default,
                                WorkingInt = default,
                                NotWorking = default,
                            };
    }
}

```

**Comment by garayx** [comment]
Hi,

I have opened an issue to fix it, 
https://issues.hibernatingrhinos.com/issue/RavenDB-20079

**Comment by oiBio** [comment]
Fixed with 5.4.110
